### DISCUSSION TITLE
Configuring product position/height on AGV

### DISCUSSION THREAD

**Discussion by jefersonflamini** [VDA5050/VDA5050#140] [Q&A]
Considering a scenario where we have AGVs in a continuous production line, where at each workstation the AGV must adjust the position of the product to a specific height (lifting table in the AGV).
This height must be set at the beginning of the edge and maintained during the movement to the next node.

I imagine that for this scenario we must have a specific Action on the node, is this correct?
Is there already an action for this type of operation?

**Comment by ralienpp** [comment]
Have you considered the `pick` and `drop` actions for this purpose? https://github.com/VDA5050/VDA5050/blob/main/VDA5050_EN.md#-682-predefined-action-definitions-description-of-their-states

They take various optional parameters, of which the relevant one is `height`.

**Comment by jefersonflamini** [comment]
Hi @ralienpp.
I believe it wouldn't be the best option, because it wouldn't be semantic and we also don't have a way to differentiate if the load/cargo should be picked/dropped or if we just have to adjust the height.
A pick/drop action must start/end with a specific load state (agv is loaded or empty) and we also couldn't check if it ended correctly.

**Reply by ralienpp** [reply]
You're right, the pick/drop actions do _not_ match the desired use case perfectly, even though they are reasonably close. The best course of action, in my opinion, would be to propose such a new action and discuss it at the next VDA5050 meeting.


>  and we also couldn't check if it ended correctly

How so? The AGV can send a state message where it sets the status of the action as either done or not done, depending on whether it was able to set the lift to the right height, no?

**Reply by jefersonflamini** [reply]
Ok, I'll commit this proposal.

I refer to the use of the pick/drop action, if you use these actions to just adjust the height, the finish status could either indicate completed adjustment or unloaded, for example.

Thanks
